lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/TaskManager.cs

```diff
@@ -362,10 +409,41 @@ private TaskStateEvent GetTaskErrorEventByExceptionType(IFailedTask failedTask)
             {
                 return TaskStateEvent.FailedTaskCommunicationError;
             }
-            else
+            if (exception is IMRUTaskSystemException)
             {
                 return TaskStateEvent.FailedTaskSystemError;
             }
+
+            // special case for communication error during group communication initialization
+            if (exception is TaskClientCodeException)
+            {
+                {
+                    // try extract cause and check whether it is InjectionException for GroupCommClient
+                    if (exception.InnerException != null &&
+                        exception.InnerException is InjectionException &&
+                        exception.InnerException.Message.Contains("GroupCommClient"))
```

Comment:
this is dangerous. What if someone changes the message in groupComm. IMRU will stop working then. In my opinion checking message content is not a good way.